### ISSUE TITLE
StackTrace with MadExcept on 64-bit builds

### ISSUE THREAD

**Issue by Daniel-Tr** [VSoftTechnologies/DUnitX#166]
Hi,

we've been struggling with call stacks created by DUnitX using MadExcept with 64-bit builds. Whilst everything looks fine with 32-bit, the stack traces on 64-bit contain only one line (the last frame).

Proposed solution (thanks to madshi for this tip) would be to use madExcept.GetCrashStackTrace instead of madStackTrace.StackTrace in TMadExcept(3|4)StackTraceProvider.GetStackTrace, at least with 64-bit builds.

Please note: This function has to be called from within the 'except'-block of a 'try .. except' to work. This limitation should be matched and thus be of no impact, according to TDUnitXTestRunner.ExecuteTests.

Best regards,
Daniel

**Comment by rmcginty** [comment]
Have you made the changes to your local source and verified that it worked in both 32 and 64 bit?  I use madExcept too, and if you can paste code that works in here, I'll be happy to implement it, test it, and put in my next pull request if everything works properly.

**Comment by Daniel-Tr** [comment]
Hi,

sure, we changed it for our local projects (and it seems to work for now). I've attached the patched file.:
[DUnitX.StackTrace.MadExcept4.zip](https://github.com/VSoftTechnologies/DUnitX/files/577609/DUnitX.StackTrace.MadExcept4.zip)

Some lines are commented out yet. They may be uncommented to use the 'old' madStackTrace.StackTrace with 32-bit builds and GetCrashStackTrace for 64-bit builds. GetCrashStackTrace will also produce good traces with 32-bit, but the traces from madStackTrace seem slightly better. (We're mostly running on 64-bit, so I wouldn't want to decide whether the compiler switch is worth the effort ..)
